lib/decorateWithMetrics.js

```diff
@@ -0,0 +1,132 @@
+import _ from 'underscore';
+
+/**
+ * Each key is a method name and the value is an array of calls metadata
+ * @type {Record<String, Array<CallMeta>>}
+ */
+let stats = {};
+
+/**
+ * Class instances are easily tracked by Dev tools memory snapshots
+ * Wrapping a result with such instance allows us to track it without
```

Comment:
I'm not quite sure what this comment means or what it implies. What is the "extra weight"?